### PR TITLE
SAML new model validation: Issuer

### LINKED ISSUE #2942: Implement and test issuer validation in SAMLTokenHandler using the new token validation model
Following the implementation in the SAML2TokenHandler, we need to add it to the other SAML token handler and add tests for it.

Part of #2711

### PR DESCRIPTION
**Pull request by iNinja**
# SAML new model validation: Issuer
- Implemented issuer validation for SAML tokens using the new validation model
- Added comparison tests to ensure the new validation model behaves like the original.

Resolves #2942 

Part of #2711

### CODE REVIEW

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs [hunk 1]
```diff
@@ -48,6 +48,19 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
                 return conditionsResult.UnwrapError().AddStackFrame(StackFrames.AssertionConditionsValidationFailed);
             }
 
+            var issuerValidationResult = await validationParameters.IssuerValidatorAsync(
```
Comment (kllysng) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs:51]: This is a not yet released class right? So this *isn't* a behavorial breaking change?

Comment (iNinja) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs:51]: That is correct. This is all internal and work in progress. There's a few more PRs coming adding functionality to this method.

Comment (kllysng) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs:51]: Thank you, makes sense

**Review by kllysng — APPROVED** [review]

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Pull request merged**